data_model/src/predicate.rs

```diff
@@ -438,16 +438,55 @@ pub mod numerical {
         }
     }
 
+    impl Copy for SemiInterval<u8> {}
+    impl Copy for SemiInterval<u16> {}
+    impl Copy for SemiInterval<u32> {}
+    impl Copy for SemiInterval<u64> {}
+
+    /// A both-inclusive range predicate
+    #[derive(Debug, Clone, Serialize, Deserialize, Encode, Decode, IntoSchema)]
+    pub struct Interval<T: Copy + Ord> {
+        /// The start of the range (inclusive)
+        start: T,
+        /// The limit of the range (inclusive)
+        limit: T,
```

Comment:
make note that if both are inclusive you'll have a problem specifying empty intervals which would otherwise be given as `start` == `limit`(`[start, limit>`) always obeying the invariant `start` <= `limit`. In the current state you would have to allow for `limit` < `start` to be able to define empty intervals